lib/beaker/dsl/helpers.rb

```diff
@@ -564,6 +564,8 @@ def with_puppet_running_on host, conf_opts, testdir = host.tmpdir(File.basename(
 
           yield self if block_given?
 
+        rescue Beaker::DSL::Assertions, Minitest::Assertion => early_assertion
```

Comment:
I tried that, and it didn't work for this.  There's another use case for this that I changed to, and that fixed the issue: https://github.com/puppetlabs/beaker/blob/master/lib/beaker/dsl/structure.rb#L112